pkg/registry/core/node/strategy_test.go

```diff
@@ -92,95 +83,43 @@ func TestDropFields(t *testing.T) {
 		node                    *api.Node
 		oldNode                 *api.Node
 		compareNode             *api.Node
-		enableNodeDynamicConfig bool
 	}{
 		{
 			name:        "nil pod cidrs",
-			node:        makeNode(nil, false, false),
+			node:        makeNode(nil, false),
 			oldNode:     nil,
-			compareNode: makeNode(nil, false, false),
+			compareNode: makeNode(nil, false),
 		},
 		{
 			name:        "empty pod ips",
-			node:        makeNode([]string{}, false, false),
+			node:        makeNode([]string{}, false),
 			oldNode:     nil,
-			compareNode: makeNode([]string{}, false, false),
+			compareNode: makeNode([]string{}, false),
 		},
 		{
 			name:        "single family ipv6",
-			node:        makeNode([]string{"2000::/10"}, false, false),
-			compareNode: makeNode([]string{"2000::/10"}, false, false),
+			node:        makeNode([]string{"2000::/10"}, false),
+			compareNode: makeNode([]string{"2000::/10"}, false),
 		},
 		{
 			name:        "single family ipv4",
-			node:        makeNode([]string{"10.0.0.0/8"}, false, false),
-			compareNode: makeNode([]string{"10.0.0.0/8"}, false, false),
+			node:        makeNode([]string{"10.0.0.0/8"}, false),
+			compareNode: makeNode([]string{"10.0.0.0/8"}, false),
 		},
 		{
 			name:        "dualstack 4-6",
-			node:        makeNode([]string{"10.0.0.0/8", "2000::/10"}, false, false),
-			compareNode: makeNode([]string{"10.0.0.0/8", "2000::/10"}, false, false),
+			node:        makeNode([]string{"10.0.0.0/8", "2000::/10"}, false),
+			compareNode: makeNode([]string{"10.0.0.0/8", "2000::/10"}, false),
 		},
 		{
 			name:        "dualstack 6-4",
-			node:        makeNode([]string{"2000::/10", "10.0.0.0/8"}, false, false),
-			compareNode: makeNode([]string{"2000::/10", "10.0.0.0/8"}, false, false),
-		},
-		{
```

Comment:
Do we want to retain some tests that ensure fields are dropped, etc? I think this could be useful since the API logic after removing the feature gate seems to have some backwards compatibility (for 1.25 nodes and under).

Reply:
The last node that can do dynamic kubelet config is 1.24 which will be out of support based on our skew policy. This was actually my question - how much we want to clean up/validate these fields or we want to simply remove any logic related to these fields?

Reply:
I don't think adding the logic to validate that these fields (e.g. pkg/apis/core/validation/validation.go) are not set is needed as it may break some automation or something. So it doesn't hurt to have some values in these fields. But should we keep the validation for them - this is a question

Reply:
Same for clearing these fields. For now I just removed the feature gate, but do we need to keep that logic of clearing the fields at all or any value is fine.

Reply:
I think I'd lean towards clearing the fields in all cases to avoid ambiguity, especially if there's no expectation that a 1.26 kube-apiserver interacts with 1.24 nodes/clients. If there's a possibility that clients expect the fields to remain set for existing nodes, or something like that, then we may need to be more careful... but overall I prefer clearing the fields because if the fields remain set it looks like dynamic config is enabled when it's actually not.

@liggitt thoughts?

Reply:
> The last node that can do dynamic kubelet config is 1.24 which will be out of support based on our skew policy.

We support running 1.24 nodes against 1.26 servers...

Reply:
looks like we dropped this feature in 1.24 kubelets, so 1.23 kubelets are the last nodes that support it

Reply:
agree we want to keep a test that ensures new nodes don't get this field set and updates of existing nodes don't newly set it

Reply:
Sorry about versions confusions, yes, I meant removed in 1.24.

Is there any guidance on validating these fields: https://github.com/kubernetes/kubernetes/blob/b208f5cd92dd54e3106c4244447721ae0c7c3462/pkg/apis/core/validation/validation.go#L5262

Should we:
- require them being not set (this may break some customers who still attempts to set these fields)
- remove any validation
- keep everything as is

Reply:
I would leave the current clear-on-create / clear-on-update-if-unset code in place, document the field is deprecated and may no longer be set in new nodes, leave the unit test in place to make sure the clearing code works, and leave it at that

Reply:
leave the validation in place if the field is non-empty